### ISSUE TITLE
Ubuntu 19.10 GTK Focus issue

### ISSUE THREAD

**Issue by odhinnsrunes** [mlabbe/nativefiledialog#79]
**Compilation Environment**

 - Ubuntu 19.10 (or 18.04):
 - GCC/Clang:
 - GCC 9.2.1 in 19.10:
 - CMake as a git submodule to my own project:
 - Have I attempted to reproduce the problem on the `devel` branch? No

**Describe the bug**

Dialog works great the first time, but the second time I try to open it, it is behind my main window. The main window is an SDL window that is borderless but in Windowed mode. Happens in both Clang and GCC builds in Ubuntu 18.04 and 19.10.

**User Description**

It's being used for an internal testing tool for a medium company ~60 employees.

**Comment by mlaggner** [comment]
also happens for me when calling from Java via LWJGL

**Comment by thp** [comment]
Still an issue with:

Ubuntu 20.04.1 LTS
UI: X11, GNOME Shell Ubuntu variant (`XDG_SESSION_DESKTOP=ubuntu`, `XDG_SESSION_TYPE=x11`)
Likely unrelated, but compiler used: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

```
% dpkg -l | egrep 'lib(gtk+-3|sdl2)-dev'
ii  libgtk-3-dev:amd64                                          3.24.20-0ubuntu1                      amd64        development files for the GTK library
ii  libsdl2-dev:amd64                                           2.0.10+dfsg1-3                        amd64        Simple DirectMedia Layer development files
```

**Comment by thp** [comment]
[SSCCE](http://sscce.org):

`makefile`

```
CFLAGS += $(shell sdl2-config --cflags)
LDLIBS += $(shell sdl2-config --libs)

CFLAGS += $(shell pkg-config --cflags gtk+-3.0)
LDLIBS += $(shell pkg-config --libs gtk+-3.0)

CFLAGS += -Inativefiledialog/src/include

all: example

example: example.c nativefiledialog/src/nfd_gtk.c nativefiledialog/src/nfd_common.c
	$(CC) -o $@ $^ $(CFLAGS) $(LDLIBS)

clean:
	rm -f example

.PHONY: all clean
```

`example.c`
```
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

#include <SDL.h>

#include "nfd.h"

int main()
{
    SDL_Init(SDL_INIT_VIDEO);

    SDL_Window *window = SDL_CreateWindow("nativefiledialog issue 79", SDL_WINDOWPOS_CENTERED,
            SDL_WINDOWPOS_CENTERED, 800, 480, SDL_WINDOW_SHOWN);

    int w, h;
    SDL_GetWindowSize(window, &w, &h);

    bool running = true;
    while (running) {
        SDL_Event e;
        while (SDL_PollEvent(&e)) {
            if (e.type == SDL_QUIT) {
                running = false;
                break;
            }
            if (e.type == SDL_MOUSEBUTTONDOWN) {
                nfdchar_t *filename;
                nfdresult_t res = NFD_OpenDialog("*.txt", NULL, &filename);
                if (res == NFD_OKAY) {
                    printf("Got filename: %s\n", filename);
                    free(filename);
                }
            }
        }
    }

    SDL_DestroyWindow(window);
    SDL_Quit();

    return 0;
}
```

Put those in the same folder, clone `nativefiledialog` as a subdirectory, then `make` then run `./example`.

Steps to reproduce:

1. Click on the (black) SDL window content, file dialog opens on top
2. Cancel file dialog (or open a file, doesn't matter)
2. Click ont he (black) SDL window content again

Expected result:

* File dialog opens on top

Actual result:

* File dialog opens "behind" the window

Additional notes:

I'm only able to reproduce this with the "Ubuntu" session type (X11).
It is not reproducible with the "Ubuntu on Wayland" (Wayland) session type.

**Comment by mlabbe** [comment]
PR merged into devel branch, fixed in 8e6bc78d95c8adc361262b8c838f2be792671ba2
